lib/commands/file-movement.js

```diff
@@ -0,0 +1,103 @@
+import _ from 'lodash';
+import log from '../logger';
+import path from 'path';
+import { fs, mkdirp, util, zip } from 'appium-support';
+
+// List of env variables, that can be expanded in path
+const KNOWN_ENV_VARS = [
+  'APPDATA', 'LOCALAPPDATA',
+  'PROGRAMFILES', 'PROGRAMFILES(X86)',
+  'PROGRAMDATA', 'ALLUSERSPROFILE',
+  'TEMP', 'TMP',
+  'HOMEPATH', 'USERPROFILE', 'PUBLIC'
+];
+const commands = {};
+
+commands.pushFile = async function pushFile (remotePath, base64Data) {
+  if (remotePath.endsWith(path.sep)) {
+    log.errorAndThrow(`It is expected that remote path points to a file and not to a folder. ` +
```

Comment:
I would prefer to throw [InvalidArgumentException](https://github.com/appium/appium-xcuitest-driver/pull/1301/files#diff-b394cf1ec7843c77eb14478b73321d02c718f7f8123b5416b6d88181627ac690R300) instead

Reply:
Changed to `InvalidaArgumentError` in all places.